### PR TITLE
Prefer fmp4 where available

### PR DESCRIPTION
**Pull request by knackebrot**
Fragmented MP4 provides noticeably lower muxing overhead compared to TS, resulting in smaller segments and lower overall bitrate. It works well (at the minimum) in MSE-enabled browsers, iOS 10+, Tizen 4+, Edge UWP and Xbox One.

```bash
ffmpeg -i big_buck_bunny_720p_surround.mp4 -c copy -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 6 -hls_segment_type mpegts -hls_playlist_type vod -hls_list_size 0 mpegts/bunny.m3u8
ffmpeg -i big_buck_bunny_720p_surround.mp4 -c copy -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 6 -hls_segment_type fmp4 -hls_fmp4_init_filename fmp4/init.mp4 -hls_playlist_type vod -hls_list_size 0 fmp4/bunny.m3u8

du mpegts fmp4
63936   mpegts
60492   fmp4

ffmpeg -i big_buck_bunny_240p_surround.mp4 -c copy -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 6 -hls_segment_type mpegts -hls_playlist_type vod -hls_list_size 0 mpegts_240/bunny.m3u8
ffmpeg -i big_buck_bunny_240p_surround.mp4 -c copy -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 6 -hls_segment_type fmp4 -hls_fmp4_init_filename fmp4_240/init.mp4 -hls_playlist_type vod -hls_list_size 0 fmp4_240/bunny.m3u8

du mpegts_240 fmp4_240
28180   mpegts_240
25500   fmp4_240
```

**Changes**
- Treat MSE-enabled browsers as supporting fmp4
- Remove webOS from fmp4 support list until confirmed it works
- Enable fmp4 by default

### CODE REVIEW

**Comment by TheNamelessWonderer** [comment]
This might want to be considered: https://github.com/jellyfin/jellyfin/issues/5081 This might cause some breakage for old iPad users with HEVC content as it can't actually play HEVC but tells the server it can.

File: src/scripts/browserDeviceProfile.js [hunk 1]
```diff
@@ -445,17 +466,14 @@ import browser from './browser';
...
         }
 
         if (canPlayH264(videoTestElement)) {
             mp4VideoCodecs.push('h264');
             hlsInTsVideoCodecs.push('h264');
-
-            if (browser.safari || browser.tizen || browser.web0s) {
-                hlsInFmp4VideoCodecs.push('h264');
-            }
+            hlsInFmp4VideoCodecs.push('h264');
```
Comment (dmitrylyzo) [inline src/scripts/browserDeviceProfile.js:476]: You can move the check here.
```suggestion
            if (canPlayHlsInFmp4()) {
                hlsInFmp4VideoCodecs.push('h264');
            }
```

File: src/scripts/browserDeviceProfile.js [hunk 2]
```diff
@@ -632,7 +650,7 @@ import browser from './browser';
         }
 
         if (canPlayHls() && options.enableHls !== false) {
-            if (hlsInFmp4VideoCodecs.length && hlsInFmp4VideoAudioCodecs.length && userSettings.preferFmp4HlsContainer() && (browser.safari || browser.tizen || browser.web0s)) {
+            if (hlsInFmp4VideoCodecs.length && hlsInFmp4VideoAudioCodecs.length && userSettings.preferFmp4HlsContainer() && canPlayHlsInFmp4()) {
```
Comment (dmitrylyzo) [inline src/scripts/browserDeviceProfile.js:653]: and `hlsInFmp4VideoCodecs` will be empty if it is not supported.
```suggestion
            if (hlsInFmp4VideoCodecs.length && hlsInFmp4VideoAudioCodecs.length && userSettings.preferFmp4HlsContainer()) {
```

File: src/scripts/browserDeviceProfile.js [hunk 3]
```diff
@@ -445,17 +466,14 @@ import browser from './browser';
         const hlsInTsVideoCodecs = [];
         const hlsInFmp4VideoCodecs = [];
 
-        if ((browser.safari || browser.tizen || browser.web0s) && canPlayHevc(videoTestElement, options)) {
+        if (canPlayHevc(videoTestElement, options)) {
```
Comment (dmitrylyzo) [inline src/scripts/browserDeviceProfile.js:469]:
```suggestion
        if (canPlayHevc(videoTestElement, options) && canPlayHlsInFmp4()) {
```

**Review by dmitrylyzo — COMMENTED** [review]
I tested only first 2 suggestions on Firefox 86

**Pull request closed without merging**